### PR TITLE
Log `own:ingestedCodeownersFile:*` events

### PR DESCRIPTION
**Pull request by cbart**
Part of #47062

Added logging of 3 events:

- `own:ingestedCodeownersFile:added`
- `own:ingestedCodeownersFile:updated`
- `own:ingestedCodeownersFile:deleted`

These were renamed from originally requested `own:owner:added` (and similar). At this point we do no support create-read-update-delete for individual owners, but we do for CODEOWNERS files.

## Test plan

https://www.loom.com/share/eb88efaabc9f48c5b34a14483adde791

### CODE REVIEW

File: enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go [hunk 1]
```diff
@@ -54,7 +58,21 @@ func (r *ownResolver) AddCodeownersFile(ctx context.Context, args *graphqlbacken
 	if err := r.db.Codeowners().CreateCodeownersFile(ctx, codeownersFile); err != nil {
 		return nil, errors.Wrap(err, "could not ingest codeowners file")
 	}
-
+	a := actor.FromContext(ctx)
+	if a.IsAuthenticated() && !a.IsMockUser() {
```
Comment (leonore) [inline enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go:62]: curious, in which cases are we a `MockUser`?

File: enterprise/cmd/frontend/internal/own/init.go [hunk 2]
```diff
@@ -25,6 +25,6 @@ func Init(
 ) error {
 	g := gitserver.NewClient()
 	o := own.NewService(g, db)
-	enterpriseServices.OwnResolver = resolvers.New(db, g, o)
+	enterpriseServices.OwnResolver = resolvers.New(db, g, o, observationCtx.Logger)
```
Comment (leonore) [inline enterprise/cmd/frontend/internal/own/init.go:28]: in insights we had a tendency to initialise scoped loggers within the resolver. what do the log lines look like when using this logger?

Comment (eseliger) [reply enterprise/cmd/frontend/internal/own/init.go:28]: +1, might be useful.

File: enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go (hunk 1 above)
Comment (eseliger) [reply enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go:62]: I never knew that was a thing :D What's it for?

File: enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go [hunk 3]
```diff
@@ -148,6 +180,21 @@ func (r *ownResolver) DeleteCodeownersFiles(ctx context.Context, args *graphqlba
...
+			deviceid.FromContext(ctx),
+			"own:ingestedCodeownersFile:deleted",
+			nil,
+			nil,
+			featureflag.GetEvaluatedFlagSet(ctx),
+			nil,
+		); err != nil {
+			r.logger.Warn("Could not log own:ingestedCodeownersFile:deleted")
+		}
+	}
```
Comment (eseliger) [inline enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go:197]: wonder if it'd make sense to pull this out into a function

```suggestion
func logBackendEvent(ctx Context.Context, logger log.Logger, db database.DB, eventName string) {
	a := actor.FromContext(ctx)
	if !a.IsAuthenticated() {
    	return
	}
		if err := usagestats.LogBackendEvent(
			db,
			a.UID,
			deviceid.FromContext(ctx),
            eventName,
			nil,
			nil,
			featureflag.GetEvaluatedFlagSet(ctx),
			nil,
		); err != nil {
			logger.Warn("Could not log own:ingestedCodeownersFile:deleted")
		}
	}
```

**Review by eseliger — APPROVED** [review]

File: enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go (hunk 1 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go:62]: It's used [here and there](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph@2539abb9f34ca37cd926379616273232a58d7ded/-/blob/internal/actor/actor.go?L63:6-63:18#tab=references).

I've seen it [in search](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3f584860ac44b7bb6d44d80109e3bb7958cdc59/-/blob/internal/search/job/jobutil/log_job.go?L140:29) while working on pings.

File: enterprise/cmd/frontend/internal/own/init.go (hunk 2 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/init.go:28]: 👀 need to test this out. In all honesty, I was copying the search pattern from [its logging site](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3f584860ac44b7bb6d44d80109e3bb7958cdc59/-/blob/internal/search/job/jobutil/log_job.go?L151), and I wired the closest logger I could find 🤭

File: enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go (hunk 3 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/resolvers/codeowners_resolvers.go:197]: 👍

File: enterprise/cmd/frontend/internal/own/init.go (hunk 2 above)
Comment (cbart) [reply enterprise/cmd/frontend/internal/own/init.go:28]: https://www.loom.com/share/974342e81e7d4ce999ffb58b16ff8d11

Before:
Screenshot 2023-03-09 at 11 50 08 AM

After `.Scope("own",...)`
Screenshot 2023-03-09 at 11 50 21 AM

I think the latter is a bit better.

Comment (cbart) [reply enterprise/cmd/frontend/internal/own/init.go:28]: I'll add the scope call here

**Pull request merged**